README.md

```diff
@@ -1,2 +1,1088 @@
+  Can be repeated to override previous occurances or to partially override a
+  preset.
++ `-g`/`--goals` - Accepts a [`position_set` argument](#the-position-set-type).
+  Can be repeated to override previous occurances or to partially override a
+  preset.
++ `-p`/`--preset` - Accepts a preset name from the `presets.json` file.
+  A preset bundles a maze, a starting position, an initial heading and a set of
+  goals.
+  Can be repeated to override previous settings.
++ `-e`/`--engine` - A UI engine.
+  Defaults to the `default` GUI Pygame-based GUI that is installed with this
+  package.
+
+  Alternative engines can be added using the [`micromouse.gui`](#adding-other-renderers)
+  entrypoint.
+
+For example, running with the `bad_deadend*.maze` mazes are compatible with the
+`semifinal` preset:
+
+```sh
+micromouse sim -p semifinal -m mazes/bad_deadends.maze
+```
+
+## Using the Simulator as a Package
+
+The simulation can also be accessed through the `sim` Python package for
+implementing your own Micromouse solving algorithms, custom simulation, or
+tools.
+
+The primary modules are:
+
+### `sim.directions`
+
+Direction enums for navigating a maze.
+
++ `Direction` - an enum for the cardinal directions (north/east/south/west).
+  Supports turning, converting to degrees/radians above the X-axis
+  (counter-clockwise, assuming East is the X-axis's positive direction),
+  converting to/from strings.
++ `RelativeDirection` - an enum for the relative directions.
+
+### `sim.maze`
+
+Utilities for representing and rendering mazes.
+
+The `Walls` enum is flags enum for managing a cell's wall configuration.
+The flag values match the [`.maz` format](#the-maz-format)'s flags.
+The `Walls` enum also supports conversion to/from `bytes` and various rotations
+and inversions.
+
+The `Maze` class represents a basic maze.
+Supports:
+
++ `empty(height: int, width: int)` - Create an empty maze with the provided
+  size. An empty maze only has its enclosing walls.
++ `full(height: int, width: int)` - Create a full maze with the provided size.
+  A full maze has all possible walls.
++ `full_from_maze(maze: Maze)` - Create a new maze of the same size with the
+  same walls (clone a maze).
++ `from_maz_file(...)`, `from_maz(...)` - Create a maze from a
+  [`.maz`](#the-maz-format) file or `bytes`-like object.
++ `from_maze_file(...)`, `from_maze_text(...)` - Create a maze from a
+  [`.maze`](#the-maze-format) file or string.
++ `from_num_file(...)` - Create a maze from a [`.num`](#the-num-format) file.
++ `from_csv_file(...)`, `from_csv(...)` - Create a maze from a
+  [`.csv`](#the-csv-format) file or string.
++ `from_file(...)` - Create a maze from any file, the format is detected based
+  on the file suffix.
++ `size` - A `(height, width)` size of the maze.
++ `height` - The number of rows in the maze.
++ `width` - The number of columns in the maze.
++ `cell_size` - The number of cells in the maze.
```

Comment:
It doesn't really matter but I find it a little confusing that we called the number of cells in the maze `cell_size`... 😅

Reply:
It's been there since the `Maze` class was created...